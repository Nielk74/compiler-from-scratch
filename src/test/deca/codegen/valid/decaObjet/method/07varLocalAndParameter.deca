// Description:
//      Test redoublant identifiers de variables locales/ parametres/ attributs.
//
// Resultats:
//    07varLocalAndParameter.expected
//
// Historique:
//    cree le 16/06/2022


class A{
    int x =1;
}

class B extends A {
    void incrementX(int x){
        this.x = this.x + x;
    }
    void increment4(){
        int x = 4;
        this.x = this.x + x;
    }
    int multiply2(){
        return 2 * x;
    }
}

{
    B b = new B();
    b.incrementX(1);
    println("x +1 = ",b.x);
    b.increment4();
    println("then x + 4 =",b.x);
    println("then x * 2 =",b.multiply2());

}