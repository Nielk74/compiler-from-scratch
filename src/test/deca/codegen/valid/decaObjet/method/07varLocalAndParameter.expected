x +1 = 2
then x + 4 =6
then x * 2 =12
